spec/erb_lint/linters/require_input_autocomplete_spec.rb

```diff
@@ -38,6 +38,82 @@
     end
   end
 
+  describe 'input field helpers linting' do
+    subject { offenses }
+
+    context 'usage of date_field_tag with autocomplete' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= date_field_tag autocomplete: "foo" do %>
+        FILE
+
+      it { expect(subject).to(eq([])) }
+    end
+
+    context 'usage of text_field_tag with autocomplete' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= text_field_tag autocomplete: "foo" do %>
+        FILE
+
+      it { expect(subject).to(eq([])) }
+    end
+
+    context 'usage of week_field_tag with autocomplete' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= week_field_tag autocomplete: "foo" do %>
+        FILE
+
+      it { expect(subject).to(eq([])) }
+    end
+
+    context 'usage of date_field_tag without autocomplete value' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= date_field_tag do %>
+        FILE
+
+      it do
+        expect(subject).to(eq([
+          build_offense(7..30,
+            "Input field helper is missing an autocomplete attribute. If no autocomplete behaviour "\
+            "is desired, use the value `off` or `nope`."),
+        ]))
+      end
+    end
+
+    context 'usage of text_field_tag without autocomplete value' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= text_field_tag do %>
+        FILE
+
+      it do
+        expect(subject).to(eq([
+          build_offense(7..30,
+            "Input field helper is missing an autocomplete attribute. If no autocomplete behaviour "\
+            "is desired, use the value `off` or `nope`."),
+        ]))
+      end
+    end
+
+    context 'usage of week_field_tag without autocomplete value' do
+      let(:file) { <<~FILE }
+        <br />
+        <%= week_field_tag do %>
+        FILE
+
+      it do
+        expect(subject).to(eq([
+          build_offense(7..30,
+            "Input field helper is missing an autocomplete attribute. If no autocomplete behaviour "\
+            "is desired, use the value `off` or `nope`."),
+        ]))
+      end
+    end
+  end
+
```

Comment:
do you intentionally not test for the other input field helpers? (url, password, etc...)

Reply:
I think having a generic test name that test a single one or a generic test name that iterate over all of them would be better because as it is it is impossible for someone reading the test code to know why we're testing those 3 and not others. (testing all of the them would be better since it would decouple the test from the implementation).